toothpick-runtime/src/main/java/toothpick/Toothpick.java

```diff
@@ -39,12 +39,31 @@
   // its transformation
   private static final ConcurrentHashMap<Object, Scope> MAP_KEY_TO_SCOPE =
       new ConcurrentHashMap<>();
+  private static final ConcurrentHashMap<Object, Scope> ROOT_SCOPES = new ConcurrentHashMap<>();
   static Injector injector = new InjectorImpl();
 
   protected Toothpick() {
     throw new RuntimeException("Constructor can't be invoked even via reflection.");
   }
 
+  /**
+   * @return The root scope as follows: <br>
+   *     <ul>
+   *       <li>If there's an existing scope tree the root scope of that tree
+   *       <li>If there's no tree it will create a default root scope and return it
+   *       <li>If there are multiple scope trees (e.g. a forest) this method will throw an exception
+   *     </ul>
+   */
+  public static Scope openRootScope() {
+    if (ROOT_SCOPES.size() > 1) {
+      throw new RuntimeException(
+          "You have multiple root scopes defined. openRootScope() will work only with a single scope tree!");
+    } else if (ROOT_SCOPES.size() == 1) {
+      return ROOT_SCOPES.values().iterator().next();
+    }
+    return openScope(Toothpick.class);
```

Comment:
I think that we could use a constant instead. That way if someone wanna add a child to the root scope the have 2 ways of using it:
1. With the new DSL: 
`openRootScope().openSubScope(this)`
2. But they can still use the one existing in TP 1.0 and 2.0, minimizing the changes on their code base:
`Toothpick.openScopes(Toothpick.ROOT_SCOPE, this)`

Reply:
I like the hidden constant that Mihaly is using and the fact the JVM ensure the unicity, as much as Android ensuresnthe unicity of the app scope.

Reply:
So that means that if someone uses the root scope before a name is assigned to it, they will have to use only the new DSL?

Reply:
I don't follow you @dlemures , this new API is totally optional and time independent.

Reply:
I mean that in order to open subscopes for the root scope we will need to use the new DSL api with the `openSubScope` method. It wont be possiblie to the same using the method `openScopes` @stephanenicolas